react-app/src/components/ProposalDetailScreen/ProposalScreenAPI.ts

```diff
@@ -0,0 +1,51 @@
+import { useCallback, useMemo } from "react";
+import {
+  ProposalDetailScreenQuery,
+  ProposalDetailScreenQueryQuery,
+  ProposalDetailScreenQueryQueryVariables,
+  ProposalDetailScreenProposalFragment as Proposal,
+} from "../../generated/graphql";
+import { useLazyGraphQLQuery } from "../../hooks/graphql";
+import { mapRequestData, RequestState } from "../../models/RequestState";
+
+export function useProposalQuery(id: string): {
+  // eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents
+  requestState: RequestState<Proposal | null>;
+  fetch: (id: string) => void;
+} {
+  const [fetch, { requestState }] = useLazyGraphQLQuery<
+    ProposalDetailScreenQueryQuery,
+    ProposalDetailScreenQueryQueryVariables
+  >(ProposalDetailScreenQuery, {
+    variables: {
+      id,
+    },
```

Comment:
Can we skip the default variable here? feels a bit redundant to initialize the hook with a proposal id and then call fetch with the same id